cnxdb/cli/main.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+"""cnx-db database control"""
+from __future__ import print_function
+import argparse
+import sys
+from functools import reduce
+
+
+def create_common_db_args_parser():
+    """Creates a ``argparse.ArgumentParser`` instance containing common
+    database arguments. These arguments are modeled after ``psql``'s
+    arguments.
+
+    """
+    parser = argparse.ArgumentParser("database arguments", add_help=False)
+    parser.add_argument('-h', '--host',
+                        default='localhost',
+                        help="database host name")
+    parser.add_argument('-p', '--port',
+                        default='5432',
+                        help="database port")
+    parser.add_argument('-d', '--dbname',
+                        help="database name")
+    parser.add_argument('-U', '--user',
+                        help="database user")
+    # parser.add_argument('-W', '--password', help="database password")
+    return parser
+
+
+def _compile_connection_string_parts(args_namespace):
+    """Given an ``argparse.Namespace``, translate this to a connection string
+    parts (dict).
+
+    """
+    return {
+        'dbname': args_namespace.dbname,
+        'user': args_namespace.user,
+        # 'password': None,
+        'host': args_namespace.host,
+        'port': args_namespace.port,
+    }
+
+
+def _translate_parts_to_string(connection_string_parts):
+    """Translate the connection string parts ot a string"""
+    return reduce(
```

Comment:
Is it the same as `' '.join(['='.join(x) for x in connection_string_parts.items() if x[1] is not None])`?